### ISSUE TITLE
handle list insertion during item submission

### ISSUE THREAD

**Issue by ael-code** [insomnia-lab/libreant#39]
see the following discussion for more info
https://github.com/insomnia-lab/libreant/pull/22#issuecomment-70121998

**Comment by ael-code** [comment]
http://sliptree.github.io/bootstrap-tokenfield/
Seems nice. But it's impossible to modify an already inserted token.

**Comment by boyska** [comment]
Yep! I prefer the "twitter typeahead" kind, where you can also use spaces. You can even edit already inserted token, just double clicking on it (or pressing your left arrow to "select" it, then pressing enter to edit it).

**Comment by g3-3k** [comment]
I ve no idea you re talkng about, please do it!

**Comment by boyska** [comment]
@g3-3k who do you refer with "you"? If that's me, then I was picking my favourite from https://sliptree.github.io/bootstrap-tokenfield/

My favourite is the second, **Using Twitter Typeahead**. Since this issue is mostly about userinterface, your opinion would be important.
Oh, why don't you try to do something about this issue? I can give you some support, if you so prefer.

**Comment by boyska** [comment]
I set 0.3 as the milestone for this just to show that I am committing in doing this. @g3-3k I will be glad to do this together with you. But I would also like to do it in the next week, so please, confirm or drop this so I can get organized.

**Comment by ael-code** [comment]
Can we postpone this to the next release?

**Comment by boyska** [comment]
Yes (and it's partially my fault that it wasn't closed).

**Comment by boyska** [comment]
@ael-code ok I started working on it. I can just say **WTF**! Can you help me in understanding the flow of it all? I'll put some issues here.

1. I would have expected that the form is validated. Indeed, it is not, see #121 
2. I can fix that, if you are ok on fixing it at the webant layer.
3. ... but even if I do, how can I get each property to _parse_ raw data to get a more meaningful data? `PresetManager.validate` does not return data ready to be pushed to archivant, which is what I would have expected. I can try to do this, but do you think it is good design? Or do you see a better way?

**Comment by boyska** [comment]
I think I have done most of it, see c1de81f39af0

Things to do:
* nice ui with typeahead
* autocompletion
